test/langtools/tools/javac/launcher/SourceLauncherTest.java

```diff
@@ -774,6 +774,50 @@ private static void markModuleAsIncubator(Path moduleInfoFile) throws Exception
             }
         }
 
+    @Test
+    public void testAbstractClassInstanceMain(Path base) throws IOException {
+        tb.writeJavaFiles(base,
+                          """
+                          public abstract class AbstractMain {
+                              void main(String[] args) {}
+                          }
+                          """);
+        testError(base.resolve("AbstractMain.java"), "",
+                "error: abstract class: AbstractMain cannot be instantiated");
+    }
+
+    @Test
+    public void testWrongMainPrivate(Path base) throws IOException {
+        tb.writeJavaFiles(base,
+                          """
+                          public class WrongMainPrivate {
+                              private static void main(String[] args) {}
```

Comment:
Same comment as in the other place, perhaps we should include a `private void main(String[] args) {}` too?

Reply:
Done:
https://github.com/openjdk/jdk/pull/25256/commits/a351686d1aeb7a82f52141558ca7d2590cb7689f